app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+cache_archive = {}
+
+
+def cache(func: Callable) -> Callable:
+    def wrapper(*args: tuple, **kwargs: dict) -> Any:
+        save_func_done = f"{func} {args} {kwargs}"
```

Comment:
if you consider the advise above then you don't need to create this string and just use args as a key